Composer/packages/adaptive-form/src/components/fields/RecognizerField/RecognizerField.tsx

```diff
@@ -1,59 +1,132 @@
 
   useMigrationEffect(value, onChange);
   const { recognizers: recognizerConfigs, currentRecognizer } = useRecognizerConfig();
-  const dropdownOptions = useMemo(() => getDropdownOptions(recognizerConfigs, shellData, shellApi), [
+  const detailsListItems = useMemo(() => getDetailsListItems(recognizerConfigs, shellData, shellApi), [
     recognizerConfigs,
   ]);
 
   const RecognizerEditor = currentRecognizer?.recognizerEditor;
   const widget = RecognizerEditor ? <RecognizerEditor {...props} /> : null;
 
-  const submit = (_, option?: IDropdownOption): void => {
-    if (!option) return;
+  const selection = useMemo(() => {
+    return new Selection({
+      onSelectionChanged: () => {
+        const selectedItems = selection.getSelection() as RecognizerListItem[];
+        if (selectedItems?.length > 0) {
+          setSelectedRecognizer(selectedItems[0]);
+        }
+      },
+    });
+  }, []);
+
+  const submit = useCallback((): void => {
+    if (!selectedRecognizer) return;
 
-    const recognizerDefinition = mapDropdownOptionToRecognizerSchema(option, recognizerConfigs);
+    const recognizerDefinition = mapListItemsToRecognizerSchema(selectedRecognizer, recognizerConfigs);
 
     const seedNewRecognizer = recognizerDefinition?.seedNewRecognizer;
     const recognizerInstance =
       typeof seedNewRecognizer === 'function'
         ? seedNewRecognizer(shellData, shellApi)
-        : { $kind: option.key as string, intents: [] }; // fallback to default Recognizer instance;
+        : { $kind: selectedRecognizer.key as string, intents: [] }; // fallback to default Recognizer instance;
     onChange(recognizerInstance);
-    telemetryClient?.track('RecognizerChanged', { recognizer: option.key as string });
-  };
+    telemetryClient?.track('RecognizerChanged', { recognizer: selectedRecognizer.key as string });
+  }, [selectedRecognizer, recognizerConfigs, shellData, shellApi, telemetryClient]);
 
   return (
     <React.Fragment>
       <FieldLabel description={description} helpLink={uiOptions?.helpLink} id={id} label={label} required={required} />
-      <Dropdown
-        ariaLabel={formatMessage('Recognizer Type')}
-        data-testid="recognizerTypeDropdown"
-        label={formatMessage('Recognizer Type')}
-        options={dropdownOptions}
-        responsiveMode={ResponsiveMode.large}
-        selectedKey={currentRecognizer?.id}
-        onChange={submit}
-      />
+      <div style={{ fontWeight: 600 }}>{formatMessage('Recognizer/Dispatch type')}</div>
+      <div css={recognizerStyle} data-testid="recognizerTypeList">
+        <span>
+          {typeof currentRecognizer?.displayName === 'function'
+            ? currentRecognizer?.displayName({})
+            : currentRecognizer?.displayName}
+        </span>
+        <span data-testid="openRecognizerDialog" style={{ color: '#0078D4' }} onClick={() => setShowDialog(true)}>
+          {formatMessage('Change')}
+        </span>
+      </div>
+      <Dialog
```

Comment:
since this dialog is blocking, it probably should allow clicking out side of it:
isClickableOutsideFocusTrap: true under modalProps